connectors/src/connectors/slack/lib/workspace_limits.ts

```diff
@@ -0,0 +1,337 @@
+
+  // List the emails of all active members in the workspace.
+  const dustAPI = new DustAPI(
+    {
+      apiKey: ds.workspaceAPIKey,
+      workspaceId: ds.workspaceId,
+    },
+    logger,
+    { useLocalInDev: true }
+  );
+
+  const activeMemberEmailsRes =
+    await dustAPI.getActiveMemberEmailsInWorkspace();
+  if (activeMemberEmailsRes.isErr()) {
+    logger.error("Error getting all members in workspace.", {
+      error: activeMemberEmailsRes.error,
+    });
+
+    throw new Error("Error getting all members in workspace.");
+  }
+
+  return activeMemberEmailsRes.value;
+}
+
+export const getActiveMemberEmailsMemoized = cacheWithRedis(
+  getActiveMemberEmails,
+  (connector: Connector) => {
+    return `active-member-emails-connector-${connector.id}`;
+  },
+  // Caches data for 15 minutes to limit frequent API calls.
+  // Note: Updates (e.g., new members added by an admin) may take up to 15 minutes to be reflected.
+  15 * 10 * 1000
+);
+
+async function getVerifiedDomainsForWorkspace(
+  connector: Connector
+): Promise<WorkspaceDomain[]> {
+  const ds = dataSourceConfigFromConnector(connector);
+
+  const dustAPI = new DustAPI(
+    {
+      apiKey: ds.workspaceAPIKey,
+      workspaceId: ds.workspaceId,
+    },
+    logger,
+    { useLocalInDev: true }
+  );
+
+  const workspaceVerifiedDomainsRes =
+    await dustAPI.getWorkspaceVerifiedDomains();
+  if (workspaceVerifiedDomainsRes.isErr()) {
+    logger.error("Error getting verified domains for workspace.", {
+      error: workspaceVerifiedDomainsRes.error,
+    });
+
+    throw new Error("Error getting verified domains for workspace.");
+  }
+
+  return workspaceVerifiedDomainsRes.value;
+}
+
+export const getVerifiedDomainsForWorkspaceMemoized = cacheWithRedis(
+  getVerifiedDomainsForWorkspace,
+  (connector: Connector) => {
+    return `workspace-verified-domains-${connector.id}`;
+  },
+  // Caches data for 15 minutes to limit frequent API calls.
+  // Note: Updates (e.g., workspace verified domains) may take up to 15 minutes to be reflected.
+  15 * 10 * 1000
+);
+
+async function isAutoJoinEnabledForDomain(
+  connector: Connector,
+  slackUserInfo: UsersInfoResponse
+): Promise<boolean> {
+  const { user } = slackUserInfo;
+  if (!user || !user.profile) {
+    return false;
+  }
+
+  const userDomain = user.profile?.email?.split("@")[1];
+  if (!userDomain) {
+    return false;
+  }
+
+  const verifiedDomains = await getVerifiedDomainsForWorkspaceMemoized(
+    connector
+  );
+
+  const isDomainAutoJoinEnabled = verifiedDomains.find(
+    (vd) => vd.domain === userDomain
+  );
+
+  return isDomainAutoJoinEnabled?.domainAutoJoinEnabled ?? false;
+}
+
+const slackMembershipAccessBlocks = {
+  autojoin_enabled: [
+    {
+      type: "section",
+      text: {
+        type: "mrkdwn",
+        text: "Access the Slack integration by joining your Dust workspace. Only members can use this feature. Click 'Join My Workspace' to get started. For help, contact a Dust admin.",
```

Comment:
The Slack integration is accessible to members of your company's Dust workspace. Click 'Join My Workspace' to get started. For help, contact an administrator.

Reply:
Fixed.